doc/developer/design/00000000_template.md

```diff
@@ -1,42 +1,96 @@
 
-## Alternatives
+Focus on the implementation of the feature.
+This is the technical part of the design.
 
-<!--
-// Similar to the Description section. List of alternative approaches considered, pros/cons or why they were not chosen
--->
+- Is it reasonably clear how the feature is implemented?
+- What dependencies does the feature have and introduce?
+- Focus on corner cases.
+- How can we test the feature and protect against regressions?
 
-## Open questions
+# Rollout
+[rollout]: #rollout
 
-<!--
-// Anything currently unanswered that needs specific focus. This section may be expanded during the doc meeting as
-// other unknowns are pointed out.
-// These questions may be technical, product, or anything in-between.
--->
+Describe what steps are necessary to enable this feature for users.
+
+## Testing
+[testing]: #testing
+
+Describe how you will test and roll out the implementation.
+When the deliverable is a refactoring, the existing tests may be sufficient.
+When the deliverable is a new feature, new tests are imperative.
+
+Describe what metrics can be used to monitor and observe the feature.
+What information do we need to expose internally, and what information is interesting to the user?
+
+Basic guidelines:
+
+* Nearly every feature requires either Rust unit tests, sqllogictest tests, or testdrive tests.
+* Features that interact with Kubernetes additionally need a cloudtest test.
+* Features that interact with external systems additionally should be tested manually in a staging environment.
+* Features or changes to performance-critical parts of the system should be load tested.
+
+## Lifecycle
+[lifecycle]: #lifecycle
+
+If the design is risky or has the potential to be destabilizing, you should plan to roll the implementation out behind a feature flag.
+List all feature flags, their behavior and when it is safe to change their value.
+Describe the [lifecycle of the feature](https://www.notion.so/Feature-lifecycle-2fb13301803b4b7e9ba0868238bd4cfb).
+Will it start as an alpha feature behind a feature flag?
+What level of testing will be required to promote to beta?
+To stable?
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+# Conclusion and alternatives
+[conclusion-and-alternatives]: #conclusion-and-alternatives
+
+- Why is the design the best to solve the problem?
+- What other designs have been considered, and what were the reasons to not pick any other?
+- What is the impact of not implementing this design?
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- What questions need to be resolved to finalize the design?
+- What questions will need to be resolved during the implementation of the design?
+- What questions does this design raise that we should address separately?
+
+# Future work
+[future-work]: #future-work
+
+Describe what work could follow from this design, which new aspects it enables, and how it might affect individual parts of Materialize.
```

Comment:
What do you think of replacing "could" with "should"? Like, asking designers to speculate about possible work seems .. prone to invention rather than answering specific questions. Or, failing that I'm less clear about the point of this section; is it good for it to be empty, is it good for it to be full of lots of random things. Is this where we would find that this is the feature that unblocks lots of other valuable features and so we should definitely do it now?